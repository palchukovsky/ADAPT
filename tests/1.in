SCOPE hello {
   DECLARE foo;
   ACCESS foo; // SUCCESS
   DECLARE bar;

   SCOPE hootie {
      DECLARE xyz;
      DECLARE foo; // SUCCESS? -- shadows hello::foo
      ACCESS bar; // SUCCESS
      ACCESS hello::foo; // SUCCESS
   }

   ACCESS birdie; // FAIL -- not declared yet
   ACCESS xyz; // FAIL -- doesnï¿½t exist
   ACCESS hootie::xyz; // SUCCESS
   ACCESS ::hello::bar; // SUCCESS
   DECLARE birdie;
}
DECLARE hello; // FAIL -- name conflict
DECLARE hello::test; // FAIL -- not allowed
ACCESS hello::foo; // SUCCESS
ACCESS ::hello::bar; // SUCCESS
ACCESS foo; // FAIL
ACCESS hello; // FAIL -- not a name you can access, because it is a SCOPE