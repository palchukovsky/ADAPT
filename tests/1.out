ERROR 13
ERROR 14
ERROR 19
ERROR 20
ERROR 23
ERROR 24
